pod-operation/src/handlers.rs

```diff
@@ -1,8 +1,52 @@
 use socketioxide::extract::{AckSender, Data, SocketRef};
 use tracing::info;
+use crate::state::{GLOBAL_STATE, State};
 
-pub fn handle_ping(socket: SocketRef, Data(data): Data<String>, ack: AckSender) {
-	info!("Received event: {:?}", data);
-	ack.send("pong").ok();
-	socket.emit("pong", "pong").ok();
+pub fn modify_state(new_value: State) {
+    if let Ok(mut state) = GLOBAL_STATE.lock() {
+        state.value = Some(new_value);
+    }
 }
+
+pub fn read_state() -> Option<State> {
+    if let Ok(state) = GLOBAL_STATE.lock() {
+        state.value.clone()
+    } else {
+        None 
+    }
```

Comment:
Question: is there a particular reason you decided use a mutex? I don't see any places where a lock is maintained for any critical section.

Reply:
Will revisit.